packages/core/src/components/Button/Button.scss

```diff
@@ -55,7 +56,8 @@
   border-color: transparent;
   color: $color-white;
 
-  &:disabled {
+  &:disabled,
+  &.ds-c-button--disabled {
     background-color: $color-gray-lighter;
```

Comment:
Do we need to have the `ds-c-button--disabled` class for each button variation? I was thinking using the disabled HTML attribute would be a better way to go if folks want to disable a button. 

>If you do disable a button, make sure it receives the disabled styling. A button element will automatically be styled as a disabled button when it has the disabled HTML attribute, but an a element will need to have the disabled modifier class applied to it.

Reply:
Yes, we would still recommend to use the `disabled` attribute for buttons, but we should still provide both options for link elements styled as a button. Adding the `ds-c-button--disabled` also makes sure the styling is consistent regardless if you are using the classname or HTML attribute. This is also consistent with `.ds-c-button--focus`, `.ds-c-button--hover` and `.ds-c-button--active`.